### PR TITLE
Perf benchmarking of mux/demux, vc router, tunneler and ci tests

### PR DESCRIPTION
**Pull request by miawangTT**
### Problem description
Perf benchmarking of mux/demux, vc router, tunneler and add ci tests for mux, demux & vc router.

### What's changed
Modified/added the test files to slow dispatch tests. 

### Checklist
- [ ] Post commit CI passes
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: .github/workflows/build-and-unit-tests.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ on:
       timeout:
         required: false
         type: number
-        default: 25
+        default: 45
```
Comment (TT-billteng) [inline .github/workflows/build-and-unit-tests.yaml:15]: That's a big jump in time for this workflow. Is it necessary that this is part of post-commit?

Comment (davorchap) [reply .github/workflows/build-and-unit-tests.yaml:15]: @ubcheema can this be one of the frequent pipelines / perf regression?

Comment (ubcheema) [reply .github/workflows/build-and-unit-tests.yaml:15]: I need to review why the run time has increased so much. Apparently. we are only adding some perf tests.

Comment (tt-rkim) [reply .github/workflows/build-and-unit-tests.yaml:15]: Any update?

File: tests/scripts/run_cpp_fd2_tests.sh [hunk 2]
```diff
@@ -144,4 +144,14 @@ if [[ $ARCH_NAME == "wormhole_b0" ]]; then
 
     # 16 TX -> 4 x 4:1 Mux -> 4:1 Mux -> 1:4 Demux -> 4 x 1:4 Demux -> 16 RX
     TT_METAL_THREADCOUNT=64 ./build/test/tt_metal/perf_microbenchmark/routing/test_mux_demux_2level
+
+    echo "Running tt-fabric perf ubench test now..."
```
Comment (tt-rkim) [inline tests/scripts/run_cpp_fd2_tests.sh:148]:
```suggestion
    echo "::notice Running tt-fabric perf ubench test now..."
```

File: tests/scripts/run_cpp_fd2_tests.sh [hunk 3]
```diff
@@ -144,4 +144,14 @@ if [[ $ARCH_NAME == "wormhole_b0" ]]; then
...
+    for max_packet_size_words in 256 512 1024 2048; do
+        run_test "./build/test/tt_metal/perf_microbenchmark/routing/test_mux_demux --max_packet_size_words $max_packet_size_words --tx_skip_pkt_content_gen 1 --rx_disable_data_check 1 --rx_disable_header_check 1 --tx_pkt_dest_size_choice 1 --check_txrx_timeout 0 --data_kb_per_tx 1048576 --tx_queue_size_bytes 65536 --rx_queue_size_bytes 131072 --mux_queue_size_bytes 65536 --demux_queue_size_bytes 131072"
+        run_test "./build/test/tt_metal/perf_microbenchmark/routing/test_vc_mux_demux --max_packet_size_words $max_packet_size_words --tx_skip_pkt_content_gen 1 --rx_disable_data_check 1 --rx_disable_header_check 1 --tx_pkt_dest_size_choice 1 --check_txrx_timeout 0 --data_kb_per_tx 1048576 --tx_queue_size_bytes 65536 --rx_queue_size_bytes 131072 --mux_queue_size_bytes 65536 --demux_queue_size_bytes 131072"
+        if ! [ $DEVICE_NAME == "N150" ]; then
```
Comment (tt-rkim) [inline tests/scripts/run_cpp_fd2_tests.sh:152]: Perhaps it's time to yank out `run_cpp_fd2_tests.sh` from `run_tests.sh`.

I would prefer if we didn't add extra environment variables that we need to pass down from the workflow. This should ideally be a command line parameter for `run_cpp_fd2_tests.sh`.

File: .github/workflows/build-and-unit-tests.yaml (hunk 1 above)
Comment (miawangTT) [reply .github/workflows/build-and-unit-tests.yaml]: Agreed. It is difficult identify the cause of ci timeout in this PR last week.

Sounds good. I removed the tunneler test from this PR.

File: tt_metal/impl/dispatch/kernels/packet_queue.hpp [hunk 4]
```diff
@@ -846,6 +846,7 @@ class packet_output_queue_state_t : public packet_queue_state_t {
             (num_words == 0)) {
             return;
         } else if (this->remote_update_network_type == DispatchRemoteNetworkType::ETH) {
+            while(eth_txq_is_busy());
```
Comment (ubcheema) [inline tt_metal/impl/dispatch/kernels/packet_queue.hpp:849]: Let's remove this.

**Review by ubcheema — APPROVED** [review]

**Pull request merged**